drivers/vulkan/rendering_device_vulkan.h

```diff
@@ -744,6 +744,8 @@ class RenderingDeviceVulkan : public RenderingDevice {
 
 	RID_Owner<UniformSet, true> uniform_set_owner;
 
+	virtual uint64_t _buffer_get_data(RID p_buffer, uint8_t **p_mem, const std::function<void *(size_t)> &p_memcpy = malloc) override; //this causes stall, only use to retrieve large buffers for saving
```

Comment:
This function should not be virtual, if the syntax is changed to write to a pointer, it should write to a pointer, and you should know the size. Then you can add the script helpers to get as other data types.